contracts/Migration/MantleTokenMigratorV2.sol

```diff
@@ -0,0 +1,401 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.15;
+
+import {ERC20} from "solmate/src/tokens/ERC20.sol";
+import {SafeTransferLib} from "solmate/src/utils/SafeTransferLib.sol";
+
+/// @title Mantle Token Migrator V2
+/// @author 0xMantle
+/// @notice Token migration contract for the BIT to MNT token migration
+contract MantleTokenMigratorV2 {
+    using SafeTransferLib for ERC20;
+
+    /* ========== STATE VARIABLES ========== */
+
+    /// @dev The address of the BIT token contract
+    address public immutable BIT_TOKEN_ADDRESS;
+
+    /// @dev The address of the MNT token contract
+    address public immutable MNT_TOKEN_ADDRESS;
+
+    /// @dev The address of the treasury contract that receives defunded tokens
+    address public treasury;
+
+    /// @dev The address of the owner of the contract
+    /// @notice The owner of the contract is initially the deployer of the contract but will be transferred
+    ///         to a multisig wallet immediately after deployment
+    address public owner;
+
+    /// @dev A mapping of addresses with the ability to approve token migrations
+    mapping(address => bool) public approvers;
+
+    /// @dev A mapping of addresses with the ability blacklist addresses from migration
+    mapping(address => bool) public blacklisters;
+
+    /// @dev A mapping of addresses blacklisted from migration
+    mapping(address => address) public blacklistedAddresses;
```

Comment:
use `mapping(address => bool) public` maybe better and more cost-economical

Reply:
lol good catch i don't know why i did that